src/plugins/vis_type_timeseries/server/lib/search_strategies/capabilities/default_search_capabilities.ts

```diff
@@ -13,26 +13,54 @@ import {
   getSuitableUnit,
 } from '../../vis_data/helpers/unit_to_seconds';
 import { RESTRICTIONS_KEYS } from '../../../../common/ui_restrictions';
+import {
+  TIME_RANGE_DATA_MODES,
+  PANEL_TYPES,
+  METRIC_AGGREGATIONS,
+  SIBLING_PIPELINE_AGGREGATIONS,
+} from '../../../../common/enums';
+import type { Panel } from '../../../../common/types';
 
 export interface SearchCapabilitiesOptions {
   timezone?: string;
   maxBucketsLimit: number;
+  panel?: Panel;
 }
 
 export class DefaultSearchCapabilities {
   public timezone: SearchCapabilitiesOptions['timezone'];
   public maxBucketsLimit: SearchCapabilitiesOptions['maxBucketsLimit'];
+  public panel?: Panel;
 
   constructor(options: SearchCapabilitiesOptions) {
     this.timezone = options.timezone;
     this.maxBucketsLimit = options.maxBucketsLimit;
+    this.panel = options.panel;
   }
 
   public get defaultTimeInterval() {
     return null;
   }
 
   public get whiteListedMetrics() {
+    if (
+      this.panel &&
+      this.panel.type !== PANEL_TYPES.TIMESERIES &&
+      this.panel.time_range_mode === TIME_RANGE_DATA_MODES.ENTIRE_TIME_RANGE
+    ) {
+      const metricAggs = Object.values<string>(METRIC_AGGREGATIONS);
+      const siblingPipelineAggs = Object.values<string>(SIBLING_PIPELINE_AGGREGATIONS);
+      const allAvailableAggs = [...metricAggs, ...siblingPipelineAggs, 'math'].reduce(
```

Comment:
what do you think about creating SPECIAL_PIPELINE_AGGREGATIONS instead of hardcoding 'math'

![image](https://user-images.githubusercontent.com/20072247/128353148-b59bd609-decf-4bfa-856e-14801dac3f9d.png)

Reply:
done